File: JavaScript/0645_set_mismatch/solution.js
function find_error_nums(nums) {
    let num_count = {};
    for (let num of nums) {
        if (num in num_count) {
            num_count[num] ++;
        } else {
            num_count[num] = 1;
        }
    }
    let result = [];
    for (let [key,value] of Object.entries(num_count)) {
        if (value > 1) {
            result.push(key);
        }
    }
    for (let i = 1; i <= nums.length; i ++) {
        if (!nums.includes(i)) {
            result.push(i);
        }
    }
    return result;
}

const nums = [1,2,2,4];
console.log(find_error_nums(nums));

File: Python/0645_set_mismatch/solution.py
def find_error_nums(nums):
    num_count = {}
    for num in nums:
        if num not in num_count:
            num_count[num] = 1
        else:
            num_count[num] += 1
    result = []
    for key,value in num_count.items():
        if value > 1:
            result.append(key)
    for num in range(1, len(nums) + 1):
        if num not in nums:
            result.append(num)
    return result

nums = [1,2,2,4]
print(find_error_nums(nums))